hadoop-ozone/dist/src/main/license/jar-report.txt

```diff
@@ -1,11 +1,10 @@
-share/ozone/lib/accessors-smart.jar
+share/ozone/lib/FastInfoset.jar
```

Comment:
Nit: Three kinds of changes to `jar-report.txt` are mixed in this PR.

1. real dependency changes (see below)
2. fixing strictly out of alphabetical order items (e.g. `hdds-rocks-native`)
3. reordering due to different `sort` behavior (lowercase/uppercase, chars other than alphabetical) across implementations

I would prefer to stick only to the first kind of change in this PR:

```diff
@@ -1,11 +1,9 @@
-share/ozone/lib/accessors-smart.jar
 share/ozone/lib/activation.jar
 share/ozone/lib/animal-sniffer-annotations.jar
 share/ozone/lib/annotations.jar
 share/ozone/lib/annotations.jar
 share/ozone/lib/aopalliance.jar
 share/ozone/lib/aopalliance-repackaged.jar
-share/ozone/lib/asm.jar
 share/ozone/lib/aspectjrt.jar
 share/ozone/lib/aspectjweaver.jar
 share/ozone/lib/awaitility.jar
@@ -156,7 +154,6 @@
 share/ozone/lib/jooq-meta.jar
 share/ozone/lib/jsch.jar
 share/ozone/lib/json-simple.jar
-share/ozone/lib/json-smart.jar
 share/ozone/lib/jsp-api.jar
 share/ozone/lib/jsr305.jar
 share/ozone/lib/jsr311-api.jar
```

Rationale: The changes are in transitive dependencies, visible only in this file (no POM change).  Reordering lines hides the real difference.

Unfortunately `dependency` check instructs to update the file mention using `update-jar-report.sh`:

```
Please update the hadoop-ozone/dist/src/main/license/bin/LICENSE.txt file with the modification
   AND execute hadoop-ozone/dist/src/main/license/update-jar-report.sh when you are ready (after a full build)
```

which re-generates from scratch, causing such mixed changes.  To avoid those, one can simply manually remove the unnecessary entries.

Reply:
Yeah... I wasn't sure how the jar report is generated. Let's see if it is happy with this.

Reply:
It sorts both old (reference) and new lists exactly to avoid false positives due to difference in `sort`:

https://github.com/apache/ozone/blob/4559ef4956cc24eb3a049cf9664322b140e6853a/hadoop-ozone/dev-support/checks/dependency.sh#L31-L32